src/application/tester.go

```diff
@@ -37,24 +44,40 @@ func TestApp(appContext types.AppContext, logger *util.Logger, mode composebuild
 	}
 	if numFailed == 0 {
 		logger.Log("All tests passed")
-		return true, nil
+		return true, false, nil
 	}
 	logger.Logf("%d tests failed", numFailed)
-	return false, nil
+	return false, false, nil
 }
 
-// TestService runs the tests for the service and return true if the tests passed
+// TestService runs the tests for the service and return true if the tests passed, if they were interrupted,
 // and an error if any
-func TestService(serviceContext types.ServiceContext, logger *util.Logger, mode composebuilder.BuildMode) (bool, error) {
+func TestService(serviceContext types.ServiceContext, logger *util.Logger, mode composebuilder.BuildMode) (bool, bool, error) {
 	logger.Logf("Testing service '%s'", serviceContext.Dir)
 	serviceTester, err := NewServiceTester(serviceContext, logger, mode)
 	if err != nil {
-		return false, err
+		return false, false, err
+	}
+
+	var isInterrupted bool
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+	go func() {
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt)
+		<-c
+		signal.Stop(c)
+		err = serviceTester.Shutdown()
+		isInterrupted = true
+		wg.Done()
+	}()
+	if isInterrupted {
+		return false, true, err
```

Comment:
Thoughts on moving this into ServiceTester? I'd prefer if `ServiceTester.Run()` captures interrupts and shuts itself down if there is one and otherwise shutdowns before returning. This would prevent this method from having to call `Shutdown` as well.